sdk/src/main/java/io/dapr/actors/runtime/ActorRuntime.java

```diff
@@ -88,18 +87,16 @@ public static ActorRuntime getInstance() {
    * Registers an actor with the runtime.
    *
    * @param clazz The type of actor.
-   * @param actorServiceFactory An optional delegate to create actor service.
+   * @param actorFactory An optional factory to create actors.
    * This can be used for dependency injection into actors.
    */
-  public <T extends AbstractActor> void RegisterActor(Class<T> clazz, Function<ActorTypeInformation, ActorService> actorServiceFactory) {
+  public <T extends AbstractActor> void RegisterActor(Class<T> clazz, ActorFactory actorFactory) {
```

Comment:
Why can't it be just a lambda?

Reply:
Can it take an ActorService factory or builder which takes actor factory.
It would allow users to override settings at ActorService layer. It's fine for now, but keep that extensibility open for changes.

Reply:
We discussed to take a configuration class instead.

Reply:
https://stackoverflow.com/questions/18400210/java-8-where-is-trifunction-and-kin-in-java-util-function-or-what-is-the-alt

Reply:
Adding @FunctionalInterface like in here:
https://github.com/spring-projects/spring-boot/blob/bf294a762a1235b942e8d6316032d131f5e94e2b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cassandra/ClusterFactory.java